timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -16,6 +16,15 @@
 
 namespace timemachine {
 
+int get_random_batch_size(int moves) {
+    int batch_size = moves;
+    const int max_noise_per_batch = 10000;
```

Comment:
I set a max multiple of random values that could be generated to avoid memory ballooning due to setting the number of proposals greater. Could increase the value or decrease it, but with 10k moves our current standard seemed like a reasonable starting point.

Reply:
Removed in https://github.com/proteneer/timemachine/pull/1225/commits/66241297d52013ea6267787fe6a29eb4588c2397 for a fixed buffer size of 10k.